Net/SMTP.php

```diff
@@ -1112,17 +1124,51 @@ protected function authGSSAPI($uid, $pwd, $authz = '')
     public function authXOAuth2($uid, $token, $authz, $conn)
     {
         $auth = base64_encode("user=$uid\1auth=$token\1\1");
+        return $this->_authOAuth('XOAUTH2', $auth, $authz, $conn);
+    }
+
+    /**
+     * Authenticates the user using the OAUTHBEARER method.
+     *
+     * @param string $uid   The userid to authenticate as.
+     * @param string $token The access token to authenticate with.
+     * @param string $authz The optional authorization proxy identifier.
+     * @param object $conn  The current object
+     *
+     * @return mixed Returns a PEAR_Error with an error message on any
+     *               kind of failure, or true on success.
+     * @since 1.9.3
+     * @see https://www.rfc-editor.org/rfc/rfc7628.html
+     */
+    public function authOAuthBearer($uid, $token, $authz, $conn)
+    {
+        $auth = base64_encode("n,a=$uid\1auth=$token\1\1");
+        return $this->_authOAuth('OAUTHBEARER', $auth, $authz, $conn);
+    }
 
-        // Maximum length of the base64-encoded token to be sent in the initial response is 497 bytes, according to
-        // RFC 4954 (https://datatracker.ietf.org/doc/html/rfc4954); for longer tokens an empty initial
+    /**
+     * Authenticates the user using the OAUTHBEARER or XOAUTH2 method.
+     *
+     * @param string $method The method (OAUTHBEARER or XOAUTH2)
+     * @param string $auth   The authentication string (base64 coded)
+     * @param string $authz  The optional authorization proxy identifier.
+     * @param object $conn   The current object
+     *
+     * @return mixed Returns a PEAR_Error with an error message on any
+     *               kind of failure, or true on success.
+     */
+    protected function _authOAuth( $method, $auth, $authz, $conn)
```

Comment:
Protected methods do not have to start with underscore, if I remember correctly.

Reply:
Here, I wanted to do explicitly a distinction with `auth*` method, this method has a different interface and is the common method for `authOAuthBearer()` & `authXOAuth2()`
as it is a protected method it can be acceptable to remove it

Reply:
I used a different method name to avoid confusion

Reply:
Yes @alecpl , i think only private methods using underscore as prefix.

@EdouardVanbelle i understand your idea. But i am with @alecpl, that we should not name it with an underscore prefix. But, to get your idea, what do you think about naming this method like `authOAuthCommon` or `authOAuthBasic`?